### DISCUSSION TITLE
Implement OAuth2 with PKCE and Keycloak in a Flutter Web app

### DISCUSSION THREAD

**Discussion by alex27riva** [Bdaya-Dev/oidc#73] [Q&A]
Hi everyone, I want to implement login with OAuth2 and PKCE in a Flutter web app, is this library suitable?
For now I'm using `openid_client`, I followed [this article](https://scribe.rip/@rangika123.kanchana/keycloak-integration-for-flutter-web-using-openid-client-with-authorization-code-flow-489afeac6e9f) but It's not working very well. I need to click the login button twice to get the token.
Does this library work well with Flutter web and is it well maintained? Where can I find some web-specific examples?
Thanks

**Comment by ahmednfwela** [comment]
Hi @alex27riva 
yes this package is fully cross platform and supported on web

check the runnable example here https://github.com/Bdaya-Dev/oidc/tree/main/packages%2Foidc%2Fexample

**Reply by ThiemoImgraben** [reply]
Hi @ahmednfwela 

The question was about the pkce procedure. Is pkce always used? I can't find anything on the subject in the documentation. Also the example is not mentioning the pkce extension.

**Comment by alex27riva** [comment]
> Hi @alex27riva yes this package is fully cross platform and supported on web
> 
> check the runnable example here https://github.com/Bdaya-Dev/oidc/tree/main/packages%2Foidc%2Fexample

Hi, I tried the example but I'm getting this error:
```
SEVERE: 2024-04-27 21:49:19.379: OidcException: Failed to handle the response from endpoint: http://localhost:8080/realms/unimi/protocol/openid-connect/userinfo, extra: {}, error: null
```
![Error](https://github.com/Bdaya-Dev/oidc/assets/5512235/fd8270d4-42ba-4206-a17d-9fbaaae4344a)

**Reply by alex27riva** [reply]
Ok, I fixed.
I had to set unsigned User Info response from Keycloak Client
![Unsigned UserInfo](https://github.com/Bdaya-Dev/oidc/assets/5512235/ca6d4903-3ff1-43db-950b-746c8f45982e)
